modules/custom/social_magic_login/src/Service/MagicUrlCreate.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Drupal\social_magic_login\Service;
+
+use Drupal\Core\Path\PathValidatorInterface;
+use Drupal\Core\Url;
+use Drupal\user\UserInterface;
+
+/**
+ * Class MagicLoginCreate.
+ */
+class MagicUrlCreate {
```

Comment:
I think we should follow Drupal Core's example and create an interface of the MagicUrl service `MagicUrlInterface` which is implemented by `MagicUrl`. This would allow others to change the way this works by swapping out the service and controllers, leaving our tokens intact.